doc/news/changes/minor/20170809ArndtMaierPelteret

```diff
@@ -0,0 +1,4 @@
+Fixed: Debugged and fixed a long standing linker error due to missing
+instantiations of ShapeInfo::reinit variants.
```

Comment:
It seems that you only add instantiations for reinit of `ShapeInfo<VectorizedArray<double>>` and `ShapeInfo<VectorizedArray<float>>`. I merely added this code a week ago in #4680. So I wouldn't really call this a long-standing linker error... ;-)